document/array.go

```diff
@@ -81,7 +84,7 @@ func (vb ValueBuffer) Iterate(fn func(i int, value Value) error) error {
 // GetByIndex returns a value set at the given index. If the index is out of range it returns an error.
 func (vb ValueBuffer) GetByIndex(i int) (Value, error) {
 	if i >= len(vb) {
-		return Value{}, ErrValueNotFound
+		return Value{}, ErrIndexOutOfRange
```

Comment:
I did it because if statement is if the given index is greater than the `len(vb)`. It seems this the only way to get a nil Value.
```
if i >= len(vb) {
		return Value{}, ErrIndexOutOfRange
	}
```
 And I thought it more pertinent to respond to the following case. 
```
{"index out of range", `{"a": {"b": [1, 2, 3]}}`, `a.b[1000]`, document.NewIntegerValue(1), ``, true},
```

Reply:
Sorry I don't understand your comment: Is this error used somewhere specific?